src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -132,12 +133,14 @@ public static void AddSemanticTokensServices(this IServiceCollection services, L
             services.AddHandlerWithCapabilities<SemanticTokensRangeEndpoint>();
             // Ensure that we don't add the default service if something else has added one.
             services.TryAddSingleton<IRazorSemanticTokensInfoService, RazorSemanticTokensInfoService>();
+
+            services.TryAddSingleton<RazorSemanticTokensLegendService>();
```

Comment:
Why `TryAddSingleton` here? Won't this type always be unique even if something else added `IRazorSemanticTokensInfoService`?

Reply:
lol, thanks, I didn't even notice, probably just IntelliCode picked that method in completion, because of the line above :)